### ISSUE TITLE
List of overranked Haskell theorem-proving Kata

### ISSUE THREAD

**Issue by DonaldKellett** [codewars/codewars.com#1772]
As concluded in #1769 , the vast majority of Kata in Idris/Agda on theorem-proving are grossly overranked which implies that similar Kata in Haskell are even more so (e.g. "A + B = B + A? Prove it!" in Haskell would be `4 kyu` tops as an Idris/Agda Kata (assuming the non-existence of `plusCommutative` and interactive theorem proving capabilities) *even under the previous ranking criteria*).  Thus, in order to prevent Codewarriors from earning an insane amount of honor and rank progress from such simple Kata, I hereby propose the following rank adjustments to the Kata mentioned below:

1. [A + B = B + A? Prove it!](
https://www.codewars.com/kata/59db393bc1596bd2b700007f), currently ranked at `1 kyu`, should be re-ranked as `5 kyu` at the very maximum.  As @ice1000 once mentioned, proving A + B = B + A is about `6 kyu` in Idris/Agda (assuming one does not "cheat" by using `plusCommutative` from the stdlib) and the task isn't *that* much more difficult in Haskell without the interactive theorem-proving capabilities of Idris/Agda.
2. [Odd + Even = Odd? Prove it!](
https://www.codewars.com/kata/599d973255342a0ce400009b) - currently ranked at `2 kyu`, this should be `6 kyu` tops.  It's even currently available in Idris which essentially makes it a joke considering how easy it is to prove theorems in Idris.  Also, as the comment section of that Kata suggests, many experienced Haskellers with no prior knowledge of theorem-proving can easily "wing" the Kata without any idea what they are actually doing in the first place.
3. [(A + B) + C = A + (B + C)? Prove it!](
https://www.codewars.com/kata/5c2fcbcba305ad2c4a91122d) currently ranked at `4 kyu` should be `6 kyu` tops (or maybe even `7 kyu`).  This thing took me about 5 minutes to solve.
4. [DNE is equivalent to PEM? Prove it!](
https://www.codewars.com/kata/5c36e827348b3e15a56edcbf) currently at `2 kyu` should be `8 kyu` instead because Agda can auto-prove it (and one can easily translate the automatically-generated Agda code to the equivalent Haskell/Idris code).

Re-ranking these Kata will probably require the help of a Codewars administrator as ordinary power users are currently forbidden from re-adjusting the rankings of approved Kata.

**Comment by ice1000** [comment]
Nope, the last one contains two proofs and one of them cannot be proved automatically.

**Comment by Voileexperiments** [comment]
Actually I'd prefer if this issue is opened by someone more proficient in functional languages and theorem proving, to give a better judgement.

**Comment by DonaldKellett** [comment]
So I just noticed that this issue ticket got labelled as Kata QA / discussion. To be honest, this issue ticket wasn't well thought out and was more of a tantrum against initial efforts by respectable members of the CW community to stabilize the rankings for theorem-proving Kata (which has since happened, at least in Coq).

Closing this issue since "DNE equivalent to PEM" (only theorem proving Kata listed above with Coq / Agda translations) has since been re-ranked and the other three are Haskell / Idris only. I think it would be fair to leave those three at the current rankings (`1`, `2` and `4 kyu` respectively) as long as they never accept Coq / Agda / Lean translations, in order to attract Haskellers with no prior knowledge of theorem proving into the theorem proving community. If anyone (perhaps more proficient in functional programming / theorem proving than I am) thinks otherwise then they are free to re-raise a similar issue and we can discuss there.
